content/post/2023-06-06-system-dependencies/index.Rmd

```diff
@@ -0,0 +1,259 @@
+---
+slug: system-dependency
+title: "System Dependencies in R Packages & Automatic Testing" 
+authors: 
+- Hugo Gruson
+date: "2023-06-06" 
+tags: 
+- package development 
+- r-package
+output: hugodown::hugo_document
+---
+
+<!-- FIXME: Use this image for social media: https://pixabay.com/fr/photos/alice-anglais-pays-des-merveilles-2902560/ -->
+
+```{r setup, include=FALSE}
+
+knitr::opts_chunk$set(fig.path = "", comment = "", warning = FALSE)
+# knitr hook to make images output use Hugo options
+knitr::knit_hooks$set(
+  plot = function(x, options) {
+    hugoopts <- options$hugoopts
+    paste0(
+      "{{<figure src=",
+      '"', x, '" ',
+      if (!is.null(hugoopts)) {
+        glue::glue_collapse(
+          glue::glue('{names(hugoopts)}="{hugoopts}"'),
+          sep = " "
+        )
+      },
+      ">}}\n"
+    )
+  }
+)
+options(crayon.enabled = FALSE)
+
+```
+
+In a previous post, we discussed about a package dependency that goes slightly beyond the normal R package ecosystem dependency: R itself.
+Today, we step even further and discuss about dependencies that have nothing to do with R: system dependencies.
+In particular, we are going to talk about system dependencies in the context of automated testing: is there anything extra to do when setting continuous integration for your package with system dependencies?
+How does it work behind the scenes?
+And how to work with edge cases?
+
+## Introduction: specifying system dependencies in R packages
+
+Before jumping right into the topic of continuous integration, let's take a moment to introduce, or remind you, how system dependencies are specified in R packages.
+You can directly jump to the next session if this concept is fresh in your memory.
+
+Let's imagine we have a package depending on The official 'Writing R Extensions' guide states [^1]:
+
+[^1]: For R history fans, this has been the case [since R 1.7.0](https://github.com/r-devel/r-svn/blob/9c46956fd784c6985867aca069b926d774602928/doc/NEWS.1#L2348-L2350).
+
+> Dependencies external to the R system should be listed in the 'SystemRequirements' field, possibly amplified in a separate README file.
+
+One important thing to note is that this field contains free text.
+As such, to refer to the same piece of software, you could write either one of the following in the package `DESCRIPTION`:
+
+``` yaml
+SystemRequirements: ExternalSoftware
+```
+
+``` yaml
+SystemRequirements: ExternalSoftware 0.1
+```
+
+``` yaml
+SystemRequirements: lib-externalsoftware
+```
+
+## The general case: everything works automagically
```

Comment:
I once read mailchimp removed the word "automagically" from their style guide but I can't find the source again nor remember the reason. :sob:

Reply:
The only reason I can imagine is that it might be unclear for non-native speakers. I can maybe add a link to https://english.stackexchange.com/questions/26618/what-is-the-purpose-of-using-the-word-automagically-when-we-already-have-auto behind the word?

Reply:
Or just take the risk to leave it as is, I feel that's better than adding a link. :thinking: